app/controllers/users_controller.rb

```diff
@@ -22,8 +30,46 @@ def create
     end
   end
 
+  def edit
+    @user = User.find_by id: params[:id]
+  end
+
+  def update
+    @user = User.find_by id: params[:id]
+    if @user.update_attributes user_params
+      flash[:success] = t ".prupdate"
+      redirect_to @user
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    User.find(params[:id]).destroy
```

Comment:
Can you brainstorm more about it
1. If it don't find the user what will happen
2. If the destroy method is called on nil, I meant if user is not found, can it it destroy/what will happen
And what are the solutions?

Reply:
If it don't find the user, @user == nil (find by) or Error
I think about add flash alert notification if user is not found
@user = User.find_by id: params[:id]
    if @user.destroy
      flash[:success] = t ".success"
    else
      flash[:danger] = t ".failed"
    end
    redirect_to users_path